lambdas/transform-parsoid/pageParser.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/PuerkitoBio/goquery"
+	"github.com/wikimedia/phoenix/common"
+)
+
+func getPageName(head *goquery.Selection) (string, error) {
+	title := head.Find("title").First().Text()
+	return title, nil
+}
+
+func getPageUrl(head *goquery.Selection) (string, error) {
+	base := head.Find("base").First()
+	url := base.AttrOr("href", "")
+	return url, nil
+}
+
+func getPageModified(head *goquery.Selection) (time.Time, error) {
+	meta := head.Find("meta[property=\"dc:modified\"]").First()
+	modifiedDate := meta.AttrOr("content", "")
+	return time.Parse(time.RFC3339, modifiedDate)
+}
+
+func getPageSourceId(head *goquery.Selection) (int, error) {
+	meta := head.Find("meta[property=\"mw:pageId\"]").First()
+	value := meta.AttrOr("content", "")
+	return strconv.Atoi(value)
+}
+
+func getPageSourceTimeUuid(head *goquery.Selection) (string, error) {
+	meta := head.Find("meta[property=\"mw:TimeUuid\"]").First()
+	value := meta.AttrOr("content", "")
+	return value, nil
+}
+
+func getPageSourceRevision(html *goquery.Selection) (int, error) {
+	about := html.AttrOr("about", "")
+	revision := about[strings.LastIndex(about, "/")+1:]
+	return strconv.Atoi(revision)
```

Comment:
This is an example of something pretty common here, where the error is propagated as-is to the caller `parseParsoidDocumentPage()` (in this case), which in turn will propagate it to `parseParsoidDocument()`, which in turn will propagate it to `main()`.  Once logged/handled there, there might not much context remaining.  In these cases, I think it makes sense to wrap the original error in something that provides that context.  See the suggestion below:

```suggestion
        r, err := strconv.Atoi(revision)
        if err != nil {
		return -1, fmt.Errorf("Unable to format "%s" as an integer revision: %w", err)
        }
	return r, nil
```